web/packages/teleport/src/Discover/Desktop/DiscoverDesktops/DiscoverDesktops.tsx

```diff
@@ -132,12 +135,12 @@ export function DiscoverDesktops(props: State) {
 
     if (foundDesktops.length) {
       for (const desktop of foundDesktops.values()) {
-        const os = desktop.labels.find(
-          label => label.name === 'teleport.dev/os'
-        ).value;
-        const osVersion = desktop.labels.find(
-          label => label.name === 'teleport.dev/os_version'
-        ).value;
+        const os =
+          desktop.labels.find(label => label.name === 'teleport.dev/os')
+            ?.value || 'unknown os';
+        const osVersion =
+          desktop.labels.find(label => label.name === 'teleport.dev/os_version')
+            ?.value || 'unknown version';
```

Comment:
i think this may be the only fix that needs to be backported. i already backported the fix for the `original` bug (which I introduced while back) which you already tested: https://github.com/gravitational/teleport/pull/21517

the other fixes on this backport, was a consequence of refactoring the `polling` logic where we took out the `timeout` feature. I did not backport the `remove timeout` to `v10` b/c as stated here https://gravitational.slack.com/archives/C04NT17MZ60/p1675892265589959

Reply:
@kimlisa there was one other aspect to the fix as well, see the new HEAD @ 0b853169e5305d0261b453dcb88880442ab3b512